js/auth.js

```diff
@@ -1,78 +1,63 @@
+/*
+* Handle authentication and permissions
+*/
 import $ from 'jquery';
+import config from 'config';
 import Notify from 'notify';
 import i18n from 'i18n';
 
-const DEFAULTS = {
-    need_role: i18n._('Role "{role}" is required')
-}
+const DEFAULT_NEED_ROLE = i18n._('Role "{role}" is required');
+
 
+export const user = config.user;
 
 /**
- * Handle authentication and permissions
+ * Build the authentication URL given the current page and an optionnal message.
  */
-class Auth {
-    /**
-     * Fetch the needed data from the page.
-     */
-    constructor() {
-        var $el = $('meta[name=current-user]');
-
-        if ($el.length) {
-            this.user = {
-                id: $el.attr('content'),
-                slug: $el.data('slug'),
-                first_name: $el.data('first_name'),
-                last_name: $el.data('last_name'),
-                roles: $el.data('roles').split(',')
-            }
-        }
+export function get_auth_url(message) {
+    const params = {next: window.location.href};
 
-        this.auth_url = $('meta[name=auth-url]').attr('content');
+    if (message) {
+        params.message = message;
     }
 
-    /**
-     * Build the authentication URL given the current page and an optionnal message.
-     */
-    get_auth_url(message) {
-        var params = {next: window.location.href};
-
-        if (message) {
-            params.message = message
-        }
+    return config.auth_url + '?' + $.param(params);
+}
 
-        return this.auth_url + '?' + $.param(params);
+/**
+ * Check if an user is authenticated
+ */
+export function need_user(message) {
+    if (!this.user) {
```

Comment:
If `this.user` is undefined, it should trigger an auth. Isn't it the expected behavior ?